usaspending_api/bulk_download/management/commands/generate_bulk_zip.py

```diff
@@ -0,0 +1,107 @@
+import logging
+import csv
+import json
+import jsonpickle
+
+from django.core.management.base import BaseCommand
+from django.conf import settings
+from usaspending_api.bulk_download.models import BulkDownloadJob
+from usaspending_api.awards.models import Award
+from usaspending_api.download.lookups import JOB_STATUS_DICT
+from usaspending_api.common.csv_helpers import sqs_queue
+from usaspending_api.bulk_download.filestreaming import csv_selection
+
+# Logging
+# logging.basicConfig(filename='bulk-download-worker.log',
+#                     format='%(levelname)s %(asctime)s :: %(message)s',
+#                     datefmt='%m/%d/%Y %I:%M:%S %p',
+#                     level=logging.INFO)
+logger = logging.getLogger('console')
+
+# # AWS parameters
+# BULK_DOWNLOAD_S3_BUCKET_NAME = os.environ.get('BULK_DOWNLOAD_S3_BUCKET_NAME')
+# BULK_DOWNLOAD_SQS_QUEUE_NAME = os.environ.get('BULK_DOWNLOAD_SQS_QUEUE_NAME')
+# BULK_DOWNLOAD_AWS_REGION = os.environ.get('BULK_DOWNLOAD_AWS_REGION')
+# DATABASE_URL = os.environ.get('DATABASE_URL')
+
+class Command(BaseCommand):
+
+    current_job_id = None
+
+    def get_current_job(self):
+        """Return the job currently stored in current_job_id"""
+        # the job_id is added to current_job_id at the beginning of the validate
+        # route. we expect it to be here now, since validate is
+        # currently the app's only functional route
+        job_id = self.current_job_id
+        if job_id:
+            return BulkDownloadJob.objects.filter(bulk_download_job_id=job_id).first()
+
+    def mark_job_status(self, job_id, status_name, skip_check=False):
+        """
+        Mark job as having specified status.
+        Jobs being marked as finished will add dependent jobs to queue.
+
+        Args:
+            job_id: ID for job being marked
+            status_name: Status to change job to
+        """
+        job = BulkDownloadJob.objects.filter(bulk_download_job_id=job_id).first()
+        # update job status
+        job.job_status_id = JOB_STATUS_DICT[status_name]
+
+
+    def handle(self, *args, **options):
+        """Run the application."""
+
+        queue = sqs_queue(region_name=settings.BULK_DOWNLOAD_AWS_REGION,
+                          QueueName=settings.BULK_DOWNLOAD_SQS_QUEUE_NAME)
+
+        logger.info('Starting SQS polling')
+        while True:
```

Comment:
How is this process killed?

Reply:
It's designed to be constantly running on an EC2 instance to handle any messages in SQS queue (which in essence are the requests for bulk downloads), generate the csvs, zip it up, and upload it to an S3 bucket. If we wanted to kill it, we could terminate the server or kill it via command line.